src/Annotation.cpp

```diff
@@ -0,0 +1,500 @@
+
+        if (targetChunks[2].find(')') != targetChunks[2].size() - 1) {
+          logout("Annotation String Error 17: invalid annotation '"
+                 << rawAnnotationString
+                 << "' target is malformed; last char of '" << targetChunks[2]
+                 << "' should be ')'") return false;
+        }
+        // targetChunks[2] is "FIELD(str)"
+      }
+    }
+  } else if (targetType == "STRUCT") {
+    if (targetChunksSize == 1) {
+      logout("Annotation String Error 25: invalid annotation '"
+             << rawAnnotationString
+             << "' target is malformed; STRUCT may be missing FIELD specifier "
+                "or specifier is malformed") return false;
+    }
+    // targetChunks looks ["STRUCT(name)", "FIELD(str)"]
+    if (!dataflow::hasOnlyOneBalancedParentheses(targetChunks[1])) {
+      logout("Annotation String Error 18: invalid annotation '"
+             << rawAnnotationString
+             << "' target is malformed; number of balanced parentheses on '"
+             << targetChunks[1] << "' does not equal one") return false;
+    }
+
+    std::string fieldSpecifier =
+        targetChunks[1].substr(0, targetChunks[1].find('('));
+    if (fieldSpecifier != "FIELD") {
+      logout("Annotation String Error 19: invalid annotation '"
+             << rawAnnotationString << "' keyword 'FIELD' missing/malformed, '"
+             << fieldSpecifier << "' found") return false;
+    }
+    // targetChunk[1] is "FIELD("
+
+    std::string fieldName =
+        dataflow::sliceString(targetChunks[1], targetChunks[1].find('(') + 1,
+                              targetChunks[1].find(')') - 1);
+    if (!dataflow::isValidCVariableName(fieldName)) {
+      logout("Annotation String Error 20: invalid annotation '"
+             << rawAnnotationString << "' FIELD argument '" << fieldName
+             << "' is not valid C variable name") return false;
+    }
+    // targetChunks[1] is "FIELD(str"
+
+    if (targetChunks[1].find(')') != targetChunks[1].size() - 1) {
+      logout("Annotation String Error 21: invalid annotation '"
+             << rawAnnotationString
+             << "' target is malformed; last character of '" << targetChunks[1]
+             << "' was not ')'") return false;
+    }
+    // targetChunks[1] is "FIELD(str)"
+  }
+  //* --- chunks[4] checker section close ---
+  if (chunks[5] != "methods") {
+    logout("Annotation String Error 22: invalid annotation '"
+           << rawAnnotationString
+           << "' missing keyword 'methods'/malformed methods "
+              "argument") return false;
+  }
+
+  if (chunks[6] != "=") {
+    logout("Annotation String Error 23: invalid annotation '"
+           << rawAnnotationString
+           << "' malformed methods argument") return false;
+  }
+
+  // chunk[7..end] should just be method args
+
+  std::string methodsString;
+  for (int i = 7; i < chunks.size(); i++) {
+
+    // removing potential "\0"'s in the string
```

Comment:
Why would there be `\0`s in the string?

Reply:
In the LLVM IR all strings end in "\00" automatically; I imagine this not being a problem once we write our own analysis on the C code but for now, we have to handle it